openqabot/pc_helper.py

```diff
@@ -100,6 +102,133 @@ def apply_publiccloud_pint_image(settings):
+    :param str state active/inactive used as name of the json file in PINT url
+    :param str query url of PINT, has to have '/' at the end
+    :param str region optional filter, mostly needed by AWS
+    :return Return dictionary has keys NAME and ID
+    """
+    settings = {}
+    try:
+        url = f"{query}{state}.json"
+        log.debug("Pint url:%s name_filter:%s region:%s", url, name_filter, region)
+        images = pint_query(url)["images"]
+        image = get_recent_pint_image(images, name_filter, region=region, state=state)
+        if image is None:
+            raise ValueError(
+                f"Cannot find matching image in PINT with name:[{name_filter}] and state:{state}"
+            )
+        if field not in image.keys() or "name" not in image.keys():
+            raise ValueError(
+                f"Cannot find expected keys '{field}' in the selected image dictionary {image}"
+            )
+        settings["ID"] = image[field]
+        settings["NAME"] = image["name"]
+    except BaseException as e:
+        log.warning(f"get_pint_image handling failed: {e}")
+    log.debug("Return settings:%s", settings)
+    return settings
+
+
+def pint_url(pint_base_url, csp_name):
+    """
+    :param str pint_base_url, usually https://susepubliccloudinfo.suse.com/v1
+    :param str csp_name microsoft, amazon, google, ...
+    :return str PINT url for the images API
+    """
+    url = "/".join([pint_base_url] + [csp_name, "images"]) + "/"
+    log.debug("PINT url:%s", url)
+    return url
+
+
+def sles4sap_pint_gce(name_filter, state, pint_base_url):
+    """
+    Query PINT about GCE images and retrieve the latest one
+    """
+    job_settings = {}
+    ret = get_pint_image(
+        name_filter=name_filter,
+        field="project",
+        state=state,
+        query=pint_url(pint_base_url, "google"),
+    )
+    if any(ret):
+        job_settings["SLES4SAP_QESAP_OS_VER"] = f"{ret['ID']}/{ret['NAME']}"
+        job_settings["SLES4SAP_QESAP_OS_STATE"] = state
+    return job_settings
+
+
+def sles4sap_pint_ec2(name_filter, state, pint_base_url, region_list):
+    """
+    Query PINT about EC2 images and retrieve the latest one
+    for each of the requested regions.
+    Returned data is organized in a different way from sles4sap_pint_azure
+    and sles4sap_pint_gce
+    """
+    job_settings = {}
+    images_list = {}
+    for this_region in region_list:
+        ret = get_pint_image(
+            name_filter=name_filter,
+            field="id",
+            state=state,
+            query=pint_url(pint_base_url, "amazon"),
+            region=this_region,
+        )
+        if any(ret):
+            images_list[this_region] = ret
+    if any(images_list):
+        # All element should have same name, just get the first
+        job_settings["SLES4SAP_QESAP_OS_VER"] = next(iter(images_list.items()))[1][
+            "NAME"
+        ]
+        job_settings["SLES4SAP_QESAP_OS_VER_STATE"] = state
```

Comment:
At least I should rename this one from `SLES4SAP_QESAP_OS_VER_STATE` to `SLES4SAP_QESAP_OS_STATE`

Reply:
why can't you use `PUBLIC_CLOUD_IMAGE_STATE` ?

Reply:
is it `state` needed by the test code? I'm not able to find any occurrence of `PUBLIC_CLOUD_IMAGE_STATE` in https://github.com/os-autoinst/os-autoinst-distri-opensuse